### PR TITLE
Idiomatic Micronaut

### PR DESCRIPTION
**Pull request by musketyr**
(Describe the goal of this PR. Mention any related Issue numbers)

At the moment, the integration still requires a lot of manual steps to make the application a first class citizen in Micronaut.

1. in the Micronaut application, one expect that the configuration is populated using the standard [chain](https://docs.micronaut.io/latest/guide/#config). `SlackConfiguration` class allows to configure `AppConfig` in a Micronaut way (env vars, system properties, command line arguments and configuration files).
2. The developers should write just the part which they are interested (the handlers or message senders) and these should be object of dependency injection. The developer can implement one of `Micronaut*Handler` classes to register handler to the `App` with all the benefits which Micronaut provides. For sending the messages, there is `MethodsClient` and `AsyncMethodsClient` ready to be used. It is populate with the bot token if available.3. 

To guarantee backward compatibility, all the created beans are marked as `@Secondary` so every existing factory will override the default implementation.

### Category (place an `x` in each of the `[ ]`)

* [ ] **bolt** (Bolt for Java)
* [x] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you agree to those rules.

### CODE REVIEW

**Comment by musketyr** [comment]
@seratch I would like to guarantee the the `MessagesClient` and `AsyncMessagesClient` is properly authenticated for the OAuth apps but I'm not certain what is the recommended way how to interact with the Slack API in OAuth mode

File: bolt/src/main/java/com/slack/api/bolt/App.java [hunk 1]
```diff
@@ -490,6 +490,8 @@ public App(AppConfig appConfig, Slack slack, List<Middleware> middlewareList) {
         this.oAuthExceptionHandler = new OAuthDefaultExceptionHandler(config());
 
         this.oAuthCallbackService = null; // will be initialized by initOAuthServicesIfNecessary()
+
+        this.status = Status.Stopped;
```
Comment (musketyr) [inline bolt/src/main/java/com/slack/api/bolt/App.java:494]: @seratch NPE from `synchronized(status)` blocks is thrown if status is not set in the controller. Do you want me to create a different PR for this change?

**Comment by musketyr** [comment]
Hi @seratch, I understand you reasons. I have moved the fix for `App#status` in a different PR. This would be a blocker in the future.

**Comment by seratch** [comment]
We appreciate your understanding. Also, thanks a lot for sending #1051 too!

**Comment by musketyr** [comment]
@seratch see https://agorapulse.github.io/micronaut-slack/

Thank you for your help!

**Comment by seratch** [comment]
@musketyr Wow, this is so cool! Congrats on the first version release 🥳

**Comment by musketyr** [comment]
@seratch thank you!
